File: Shift.Infrastructure/Models/ViewModels/Journals/UJournal.cs
using System.Collections.Generic;

namespace Shift.Infrastructure.Models.ViewModels.Journals
{
	using Shift.Infrastructure.Models.ViewModels.Journals.UJournalData;

	public class UJournal
	{
		public int Id { get; set; }

		public int? UndergraduateId { get; set; }

		public virtual PreparationInfoVM PreparationInfo { get; set; }
		public virtual ThesisCertificationVM ThesisCertification { get; set; }

		public virtual ICollection<ReportVM> ReportResults { get; set; } = new List<ReportVM>();
	}
}

File: Shift.Web/ClientApp/src/app/infrastracture/entities/ujournal/UJournal.ts
import { PreparationInfo } from "./PreparationInfo";
import { ThesisCertification } from "./ThesisCertification";
import { Report } from "./Report";

export class UJournal {
    Id: number;
    UndergraduateId?: number;

    PreparationInfoId: number;
    PreparationInfo: PreparationInfo;
    
    ThesisCertificationId: number;
    ThesisCertification: ThesisCertification;
    ReportResults: Report[];
}
